stripe-crypto-onramp/src/main/java/com/stripe/android/crypto/onramp/OnrampCoordinator.kt

```diff
@@ -0,0 +1,134 @@
+package com.stripe.android.crypto.onramp
+
+import androidx.activity.ComponentActivity
+import androidx.activity.result.ActivityResultRegistryOwner
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import com.stripe.android.link.LinkController
+import javax.inject.Inject
+
+/**
+ * Coordinator interface for managing the Onramp lifecycle, Link user checks,
+ * and authentication flows.
+ */
+internal interface OnrampCoordinator {
+
+    /**
+     * Initialize the coordinator with the provided configuration.
+     *
+     * @param configuration The OnrampConfiguration to apply.
+     * @param callback Callback receiving success or failure.
+     */
+    fun configure(
+        configuration: OnrampConfiguration,
+        callback: ConfigurationCallback
+    )
+
+    /**
+     * Check if the given email corresponds to an existing Link user.
+     *
+     * @param email The email address to look up.
+     */
+    fun isLinkUser(email: String)
+
+    /**
+     * Given the required information, registers a new Link user.
+     *
+     * @param info The LinkInfo for the new user.
+     */
+    fun registerNewLinkUser(info: LinkUserInfo)
+
+    /**
+     * Authenticate an existing Link user via email.
+     *
+     * @param email The email address of the existing user.
+     */
+    fun authenticateExistingLinkUser(email: String)
+
+    /**
+     * Present UI to authenticate a Link user.
+     *
+     * @param email The email address to authenticate.
+     */
+    fun presentForAuthentication(email: String)
+}
+
+/**
+ * Default implementation of [OnrampCoordinator].
+ *
+ * @param viewModel The ViewModel that persists configuration state across process restarts.
+ * @param activityResultRegistryOwner Host providing ActivityResultRegistry for LinkController.
+ * @param isLinkUserCallback Callback invoked with the result of determining if a
+ *                           provided email is associated with a Link user.
+ */
+internal class DefaultOnrampCoordinator @Inject constructor(
```

Comment:
nit - let's move this to its own file so that integrators just see the _exposed part_ when they look into this file.